qa/evil-tests/src/test/java/org/elasticsearch/plugins/ListPluginsCommandTests.java

```diff
@@ -66,18 +89,66 @@ public void testNoPlugins() throws Exception {
-        assertTrue(terminal.getOutput(), terminal.getOutput().contains("fake"));
+        assertEquals(terminal.getOutput(), buildMultiline("fake"));
     }
 
     public void testTwoPlugins() throws Exception {
         Environment env = createEnv();
-        Files.createDirectory(env.pluginsFile().resolve("fake1"));
-        Files.createDirectory(env.pluginsFile().resolve("fake2"));
+        buildFakePlugin(env, "fake desc", "fake1", "org.fake");
+        buildFakePlugin(env, "fake desc 2", "fake2", "org.fake");
         MockTerminal terminal = listPlugins(env);
+        assertEquals(terminal.getOutput(), buildMultiline("fake1", "fake2"));
+    }
+    
+    public void testPluginWithVerbose() throws Exception {
+        Environment env = createEnv();
+        buildFakePlugin(env, "fake desc", "fake_plugin", "org.fake");
+        String[] params = { "-v" };
+        MockTerminal terminal = listPlugins(env, params);
+        assertEquals(terminal.getOutput(), buildMultiline("Plugins directory: " + env.pluginsFile(), "fake_plugin",
+        		"- Plugin information:", "Name: fake_plugin", "Description: fake desc", "Version: 1.0", " * Classname: org.fake"));
+    }
+    
+    public void testPluginWithVerboseMultiplePlugins() throws Exception {
+        Environment env = createEnv();
+        buildFakePlugin(env, "fake desc 1", "fake_plugin1", "org.fake");
+        buildFakePlugin(env, "fake desc 2", "fake_plugin2", "org.fake2");
+        String[] params = { "-v" };
+        MockTerminal terminal = listPlugins(env, params);
+        assertEquals(terminal.getOutput(), buildMultiline("Plugins directory: " + env.pluginsFile(),
+        		"fake_plugin1", "- Plugin information:", "Name: fake_plugin1", "Description: fake desc 1", "Version: 1.0",
+        		" * Classname: org.fake", "fake_plugin2", "- Plugin information:", "Name: fake_plugin2",
+        		"Description: fake desc 2", "Version: 1.0", " * Classname: org.fake2"));
+    }
+        
+    public void testPluginWithoutVerboseMultiplePlugins() throws Exception {
+        Environment env = createEnv();
+        buildFakePlugin(env, "fake desc 1", "fake_plugin1", "org.fake");
+        buildFakePlugin(env, "fake desc 2", "fake_plugin2", "org.fake2");
+        MockTerminal terminal = listPlugins(env, new String[0]);
         String output = terminal.getOutput();
-        assertTrue(output, output.contains("fake1"));
-        assertTrue(output, output.contains("fake2"));
+        assertEquals(output, buildMultiline("fake_plugin1", "fake_plugin2"));
+    }
+    
+    public void testPluginWithoutDescriptionFile() throws Exception{
+        Environment env = createEnv();
+        Files.createDirectories(env.pluginsFile().resolve("fake1"));
+        NoSuchFileException e = expectThrows(NoSuchFileException.class, () -> {
+        	listPlugins(env);
+        });
+        assertTrue(e.getMessage(), e.getMessage().contains("fake1\\plugin-descriptor.properties"));
```

Comment:
I think there is a bug here. What is `\\`? I guess Windows? You need to take caution for different filesystems.